### PR TITLE
Fix currentTarget not having a value when used in the modal.

### PR DESCRIPTION
**Pull request by Minglee01**
@incuna/js 
Please review.

### CODE REVIEW

**Comment by wytrych** [comment]
Looks ok

**Comment by henrahmagix** [comment]
I'm not sure how this has changed anything. `e` is already available to the lexical scope of the controller function, and I can't see anything that would change the `e` object by the time the modal controller is initialised.

**Comment by Minglee01** [comment]
well it didnt work untill we assigned it to a variable.  Graham said it maight have to do with timing issues.

**Comment by Minglee01** [comment]
We couldnt see why it wouldnt work either
it just didnt.

File: src/angular-external-link-interceptor.js [hunk 1]
```diff
@@ -78,6 +78,9 @@
                 externalModal: function (e, href) {
                     e.preventDefault();
 
+                    // Because the model is opened later the currentTarget mey get set to null. 
+                    var currentTarget = e.currentTarget;
+
```
Comment (henrahmagix) [inline src/angular-external-link-interceptor.js]: We could use the same process for getting `href` into the controller function without problems: use the `resolve` object.

```js
resolve: {
    externalUrl: function () {
        return href;
    },
    currentTarget: function () {
        return e.currentTarget;
    }
},
controller: [
    '$scope', '$modalInstance', 'externalUrl', 'currentTarget',
    function ($scope, $modalInstance, externalUrl, currentTarget) {}
]
```

Comment (henrahmagix) [inline src/angular-external-link-interceptor.js]: @Minglee01 It'd be great to know if the above produces the same bug as before.

**Comment by grahamgilchrist** [comment]
@henrahmagix I think it's because `$modal.open()` is an a asynchronous call, the code in the controller runs at a later point than during the event callback. Some of the properties of `e`, particularly `e.currentTarget` are assigned back to `null` when the event callbacks finish. 

So I think what is happening is the event is calling `$modal.open()` asynchronously, but then the callback is completing before the modal controller code runs, and thus `e.currentTarget` is null in that code. The fix is to cache the value during the event callback in another variable.

**Pull request merged**